### PR TITLE
fix: populate start_time in request

### PR DESCRIPTION
**Pull request by ryanpq**
## Description
Fix for start time not checked in request body

Addresses: https://github.com/openfga/sdk-generator/issues/483

## References
Also backported to sdk-generator: https://github.com/openfga/sdk-generator/pull/484

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -1230,9 +1230,12 @@ func (client *OpenFgaClient) ReadChangesExecute(request SdkClientReadChangesRequ
 		req = req.ContinuationToken(*continuationToken)
 	}
 	requestBody := request.GetBody()
-	if requestBody != nil {
+	if requestBody.Type != "" {
 		req = req.Type_(requestBody.Type)
 	}
+	if !requestBody.StartTime.IsZero() {
+		req = req.StartTime(requestBody.StartTime)
+	}
```
Comment (ewanharris) [inline client/client.go]: We'll need to guard against `requestBody` being nil here as it's valid to call `ReadChanges` with no body argument. Might be worth also adding a test for a nil body now that we're expanding this
```suggestion
	if requestBody != nil &&  requestBody.Type != "" {
		req = req.Type_(requestBody.Type)
	}
	if requestBody != nil &&  !requestBody.StartTime.IsZero() {
		req = req.StartTime(requestBody.StartTime)
	}
```

**Review by ewanharris — APPROVED** [review]
